coil-svg/src/main/java/coil/decode/SvgDecoder.kt

```diff
@@ -0,0 +1,32 @@
+@file:Suppress("DEPRECATION", "unused")
+
+package coil.decode
+
+import android.graphics.drawable.PictureDrawable
+import coil.bitmappool.BitmapPool
+import coil.size.PixelSize
+import coil.size.Size
+import com.caverock.androidsvg.RenderOptions
+import com.caverock.androidsvg.SVG
+import com.caverock.androidsvg.SVGParseException
+import okio.BufferedSource
+
+/**
+ * A [Decoder] that uses android-SVG to decode SVG files.
+ */
+class SvgDecoder : Decoder {
+
+    override fun handles(source: BufferedSource, mimeType: String?) = mimeType == "image/svg+xml"
+
+    override suspend fun decode(pool: BitmapPool, source: BufferedSource, size: Size, options: Options): DecodeResult {
+        try {
+            val svg = source.use { SVG.getFromInputStream(it.inputStream()) }
+            return DecodeResult(
+                    drawable = PictureDrawable(svg.renderToPicture()),
+                    isSampled = false
+            )
+        } catch (e: SVGParseException) {
+            throw IllegalStateException("Failed to load SVG.", e)
```

Comment:
How come you're wrapping the exception here?

Reply:
I've been using this (actually a Glide SVG decoder) in my app for a while and wrap the exception to avoid needing to expose the SVG library as an API dependency.  Also because consumers don't need to know whether it's an SVG that's being loaded to handle an exception (passing a string from an API to the lib, consumers shouldn't need to know or care the file type).

That being said, let me know if you want it changed.

Reply:
Good point about keeping the decoder as an `implementation` dependency. Looks like `SVGParseException` extends `SAXException` (XML parse exception) w/o any extra metadata so I think listeners can listen for that and we can avoid wrapping the exception. It might make sense to add a `DecodeException` sometime in the future.